app/scripts/modules/core/src/serverGroup/ServerGroupHeader.tsx

```diff
@@ -0,0 +1,154 @@
+import * as React from 'react';
+import { get } from 'lodash';
+
+import { NgReact, ReactInjector } from 'core/reactShims';
+import { Application } from 'core/application';
+import { IServerGroup } from 'core/domain';
+import { JenkinsViewModel } from 'core/serverGroup/ServerGroup';
+import { EntityNotifications } from 'core/entityTag/notifications/EntityNotifications';
+import { HealthCounts } from 'core/healthCounts';
+import { CloudProviderLogo } from 'core/cloudProvider';
+import { LoadBalancersTagWrapper } from 'core/loadBalancer';
+import { ServerGroupManagerTag } from 'core/serverGroupManager/ServerGroupManagerTag';
+import { ISortFilter } from 'core/filterModel';
+import { Overridable } from 'core/overrideRegistry';
+
+export interface IServerGroupHeaderProps {
+  application: Application;
+  images?: string;
+  isMultiSelected: boolean;
+  jenkins: JenkinsViewModel;
+  serverGroup: IServerGroup;
+  sortFilter: ISortFilter;
+}
+
+export class LoadBalancers extends React.Component<IServerGroupHeaderProps> {
+  public render() {
+    const { application, serverGroup } = this.props;
+    const hasLoadBalancer = !!get(serverGroup, 'loadBalancers.length') || !!get(serverGroup, 'targetGroups.length');
+    return hasLoadBalancer &&
+      <LoadBalancersTagWrapper key="lbwrapper" application={application} serverGroup={serverGroup}/>;
+  }
+}
+
+export class MultiSelectCheckbox extends React.Component<IServerGroupHeaderProps> {
+  public render() {
+    const { isMultiSelected, sortFilter: { multiselect } } = this.props;
+    return multiselect && <input type="checkbox" checked={isMultiSelected}/>;
```

Comment:
It's probably worth adding a comment here why this checkbox has a `checked` state but no event handlers (it tripped me up looking at it just now).

(The reason it doesn't have any event handlers is because there's a click handler on an ancestor div that sets the `isMultiSelected` state).